src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -32,120 +34,61 @@
-        return new StrimziKafkaContainer(brokerId, false, additionalKafkaConfiguration)
-            .withExternalZookeeper(connectString);
-    }
-
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @param useKraft whether the broker should use Kraft
-     * @param additionalKafkaConfiguration additional configuration for Kafka broker
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithAdditionalConfiguration(final int brokerId, final boolean useKraft, final Map<String, String> additionalKafkaConfiguration) {
-        return new StrimziKafkaContainer(brokerId, useKraft, additionalKafkaConfiguration);
-    }
-
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @param additionalKafkaConfiguration additional configuration for Kafka broker
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithAdditionalConfiguration(final int brokerId, final Map<String, String> additionalKafkaConfiguration) {
-        return createWithAdditionalConfiguration(brokerId, false, additionalKafkaConfiguration);
+    private int brokerId;
+    private String kafkaVersion;
+    private String strimziTestContainerVersion;
+    private boolean useKraft;
+    private String storageUUID;
+
+    static {
+        LOGICAL_KAFKA_VERSION_ENTITY = new KafkaVersionService();
     }
 
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with empty additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @return instance of StrimziKafkaContainer
-     */
-    public static StrimziKafkaContainer create(final int brokerId) {
-        return new StrimziKafkaContainer(brokerId, false, Collections.emptyMap());
-    }
-
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with empty additional configuration of Kafka broker, and use Kraft.
-     * @param brokerId broker id
-     * @return instance of StrimziKafkaContainer
-     */
-    public static StrimziKafkaContainer createWithKraft(final int brokerId) {
-        return new StrimziKafkaContainer(brokerId, true, Collections.emptyMap());
+    private void buildDefaults() {
+        String strimziTestContainerVersion;
+        if (this.strimziTestContainerVersion == null || this.strimziTestContainerVersion.isEmpty()) {
+            strimziTestContainerVersion = LOGICAL_KAFKA_VERSION_ENTITY.latestRelease().getStrimziTestContainerVersion();
+            LOGGER.info("You did not specify Strimzi test container version. Using latest release:{}", strimziTestContainerVersion);
```

Comment:
I think using You in log messages is a bit unconventional: "No Strimzi test container version specified" would be better.